### PR TITLE
Update pillar.example for JDK1.8.0.144; fix #36

### LINKED ISSUE #36: Downloading Java Fails
I'm on Ubuntu 16.04 and when this formula runs does not download the full file, but it does exit 0

https://github.com/saltstack-formulas/sun-java-formula/blob/master/sun-java/init.sls#L27

eg `curl -L -s -o '/usr/share/java/jdk-8u144-linux-x64.tar.gz' 'http://download.oracle.com/otn-pub/java/jdk/8u144-b01/090f390dda5b47b9b721c7dfaa008135/jdk-8u144-linux-x64.tar.gz'`

This causes the untar/gzip to fail.  I was able to successfully download the file via:

`wget --no-cookies --no-check-certificate --header "Cookie: oraclelicense=a" http://download.oracle.com/otn-pub/java/jdk/8u144-b01/090f390dda5b47b9b721c7dfaa008135/jdk-8u144-linux-x64.tar.gz
--2017-08-03 22:41:59--  http://download.oracle.com/otn-pub/java/jdk/8u144-b01/090f390dda5b47b9b721c7dfaa008135/jdk-8u144-linux-x64.tar.gz`

Does anyone know a curl command that will work (ie accept the license)?  Figured that would be a simpler tweak, but if not I can refactor to use wget

### PR DESCRIPTION
**Pull request by noelmcloughlin**
This PR is minor
- Following  #38 we can align pillar.example version too
- Add missing pillar examples
- Address #36 

My derivative pillar.example was tested on Fedora/SUSE/Ubuntu.

### CODE REVIEW

**Comment by noelmcloughlin** [comment]
@blbradley Hey can do the needful again, thanks.

**Comment by blbradley** [comment]
I suppose I should have been more clear. You should file a PR for a single issue. Please do not fix multiple issues in the same PR as it can lead to discussions which are hard to follow.

I'm fine with this one, but you really _must_ break your changes into smaller chunks if you want contributors to be confident in merging changes you make.

Thanks for your time and patience!
